sklearn/tree/tests/test_tree.py

```diff
@@ -2417,6 +2417,81 @@ def test_missing_values_on_equal_nodes_no_missing(criterion):
     assert_allclose(y_pred, [np.mean(y_equal[-4:])])
 
 
+@pytest.mark.parametrize("seed", range(3))
+@pytest.mark.parametrize("criterion", ["squared_error", "friedman_mse"])
+def test_missing_values_random_splitter_etc_on_equal_nodes_no_missing(criterion, seed):
+    """Check missing values go to the correct node during predictions for ExtraTree.
+
+    Since ETC use random splits, we use different seeds to verify that the
+    left/right node is chosen correctly when the splits occur.
+    """
+    X = np.array([[0, 1, 2, 3, 8, 9, 11, 12, 15]]).T
+    y = np.array([0.1, 0.2, 0.3, 0.2, 1.4, 1.4, 1.5, 1.6, 2.6])
+
+    dtc = ExtraTreeRegressor(random_state=seed, max_depth=1, criterion=criterion)
+    dtc.fit(X, y)
+
+    # see which node has the most data points
+    dtc.tree_.value
+
+    # Get the left and right children of the root node
+    left_child = dtc.tree_.children_left[0]
+    right_child = dtc.tree_.children_right[0]
+
+    # Get the number of samples for the left and right children
+    left_samples = dtc.tree_.weighted_n_node_samples[left_child]
+    right_samples = dtc.tree_.weighted_n_node_samples[right_child]
+    went_left = left_samples > right_samples
+
+    # predictions
+    y_pred_left = dtc.tree_.value[left_child][0]
+    y_pred_right = dtc.tree_.value[right_child][0]
+
+    # Goes to node with the most data points
+    y_pred = dtc.predict([[np.nan]])
+    if went_left:
+        assert_allclose(y_pred_left, y_pred)
+    else:
+        assert_allclose(y_pred_right, y_pred)
+
+
+@pytest.mark.parametrize("criterion", ["squared_error", "friedman_mse"])
+def test_missing_values_random_splitter_on_equal_nodes_no_missing(criterion):
+    """Check missing values go to the correct node during predictions.
+
+    This checks for the case where there are no missing values during training,
+    and missing values are only present during prediction.
+    """
+    X = np.array([[0, 1, 2, 3, 8, 9, 11, 12, 15]]).T
+    y = np.array([0.1, 0.2, 0.3, 0.2, 1.4, 1.4, 1.5, 1.6, 2.6])
+
+    dtc = ExtraTreeRegressor(random_state=42, max_depth=2, criterion=criterion)
+    dtc.fit(X, y)
+
+    # Goes to right node because it has the most data points
+    decision_path = dtc.decision_path([[np.nan]])
+
+    # Note: when max_leaf_nodes is not defined, the implementation will build
+    # the tree using DFS
+    # For a 7-node binary tree with DFS traversal, this results in the following
+    # decision-path of the nodes, which is the right-side view of the binary tree
+    expected_decision_path = [[1, 0, 0, 0, 1, 0, 1]]
```

Comment:
Is there somehow we can specify what we are checking here exactly? Because there isn't clarity on what the tree is exactly just from looking at this test.

Reply:
I will remove and update the PR

Reply:
Sounds good. Thanks!

Reply:
Removed this unit test as it is not really useful in [36e7c10](https://github.com/scikit-learn/scikit-learn/pull/27966/commits/36e7c109908ebb77ff2626512fdf34e31e4c6d4c)